compyle/array.py

```diff
@@ -372,6 +368,85 @@ def arange(start, stop, step, dtype=np.int32, backend='cython'):
     return wrap_array(out, backend)
 
 
+def linspace(start, stop, num, dtype=np.float64, backend='opencl',
+             endpoint=True):
+    if not type(num) == int:
+        raise Exception("num should be an integer")
+    if num <= 0:
+        raise Exception('num should be greater than 0')
+    if backend == 'opencl':
+        import pyopencl.array as gpuarray
+        from .opencl import get_queue
+        if endpoint:
+            delta = (stop-start)/(num-1)
+        else:
+            delta = (stop-start)/num
+        out = gpuarray.arange(get_queue(), 0, num,
+                              1, dtype=dtype)
+        out = out * delta+start
+    elif backend == 'cuda':
+        import pycuda.gpuarray as gpuarray
+        import pycuda.autoinit
+        if endpoint:
+            delta = (stop-start)/(num-1)
+        else:
+            delta = (stop-start)/num
+        out = gpuarray.arange(0, num, 1, dtype=dtype)
+        out = out*delta+start
+    else:
+        out = np.linspace(start, stop, num,
+                          endpoint=endpoint, dtype=dtype)
+    return wrap_array(out, backend)
+
+
+@annotate
+def nDiff(i, y, x, b, lb):
```

Comment:
Can you change the function name to be consistent with the rest, i.e. use `lower_case_with_underscores`.